sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -237,6 +229,9 @@ case class Divide(left: Expression, right: Expression)
     }
   }
 
+  // Used by doGenCode
+  protected def divide(eval1: ExprCode, eval2: ExprCode, javaType: String): String
```

Comment:
I don't think we need this abstraction. [this one](https://github.com/apache/spark/pull/14036/files#diff-1516b10738479bbe190fb4e239258473L252) already covers both fraction and integral